spec/n_plus_one_control/executor_spec.rb

```diff
@@ -16,40 +16,54 @@
       .to raise_error(ArgumentError, "Block is required!")
   end
 
-  it "returns correct counts for default scales" do
+  it "returns correct counts for default scales", :aggregate_failures do
     result = described_class.new(population: populate).call(&observable)
 
     expect(result.size).to eq 2
     expect(result.first[0]).to eq 2
-    expect(result.first[1].size).to eq 3
+    expect(result.first[1][:db].size).to eq 3
```

Comment:
Such deep nesting may be a sign that it'd be better to return structs/objects instead of hashes. WDYT?